### PR TITLE
ec2 subnet lookup by tags module

### PR DESCRIPTION
**Pull request by pmazurek**
This is a helper module that allows for very agile parameterising of subnets whithout the need to hardcode subnet id's in the variables.

It is extremely helpful if you have to manage multiple VPC's with complex subnet layouts.

### CODE REVIEW

File: cloud/amazon/ec2_subnet_lookup.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+# You should have received a copy of the GNU General Public License
+# along with this library.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: ec2_subnet_lookup
+short_description: Get the subnets ID's by tags.
+description:
+    - This module allows you to query for subnet ID's using tags. It is useful if you have to manage complex subnet structure and use modules that rely on subnet ID's given as comma separated values.
+author: Piotr Mazurek
```
Comment (ghost) [inline cloud/amazon/ec2_subnet_lookup.py:22]: You should add the version of Ansible this module is being added to.

``` yaml
version_added: "2.0"
```

File: cloud/amazon/ec2_subnet_lookup.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    argument_spec.update(dict(
+            tags=dict(default=None, type='dict'),
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+    )
+
+    state = module.params.get('state')
```
Comment (ghost) [inline cloud/amazon/ec2_subnet_lookup.py:93]: Is this needed?

File: cloud/amazon/ec2_subnet_lookup.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    module.exit_json(changed=False, subnet_ids=subnet_ids)
+
+
+# this is magic, see lib/ansible/module_common.py
+#<<INCLUDE_ANSIBLE_MODULE_COMMON>>
+
+from ansible.module_utils.basic import *
+from ansible.module_utils.ec2 import *
+
+main()
```
Comment (ghost) [inline cloud/amazon/ec2_subnet_lookup.py:128]: This should be:

``` python
if __name__ == '__main__':
    main()
```

See http://docs.ansible.com/ansible/developing_modules.html#module-checklist

File: cloud/amazon/ec2_subnet_lookup.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        }
+    }
+}
+
+Now this can be used in other modules that require subnet ID's as CSV's:
+  some_module_value: {{subnets.subnet_ids | join(',')}}
+
+'''
+
+import sys
```
Comment (ghost) [inline cloud/amazon/ec2_subnet_lookup.py:77]: Is this needed?

File: cloud/amazon/ec2_subnet_lookup.py (hunk 1 above)
Comment (ghost) [inline cloud/amazon/ec2_subnet_lookup.py:22]: You should also add your github ID to the author line.

``` yaml
author: Piotr Mazurek (@pmazurek)
```

File: cloud/amazon/ec2_subnet_lookup.py [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    ec2_url, aws_access_key, aws_secret_key, region = get_ec2_creds(module)
+    tags = module.params.get('tags')
+
+    if region:
+        try:
+            vpc_conn = boto.vpc.connect_to_region(
+                region,
+                aws_access_key_id=aws_access_key,
+                aws_secret_access_key=aws_secret_key
+            )
```
Comment (ghost) [inline cloud/amazon/ec2_subnet_lookup.py:104]: You can make your module a little friendlier by supporting the boto profile option.

Replace your call to get_ec2_creds with:

``` python
region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module)
```

Then replace this call with:

``` python
vpc_conn = boto.vpc.connect_to_region(
    region,
   **aws_connect_kwargs
)
```

This way your module will accept AWS keys or boto profile name. We use nothing but boto profiles in my shop, which is how I discovered your module was not accepting them.

**Comment by wimnat** [comment]
I just saw this and i see it is pretty much the same as my PR https://github.com/ansible/ansible-modules-extras/pull/783.  Just thought i'd make a note for the core team.

**Comment by pmazurek** [comment]
I'm closing my PR, as #783 looks better.

**Comment by wimnat** [comment]
Thanks @pmazurek. Can you +1 the other PR to help it get merged?

**Pull request closed without merging**